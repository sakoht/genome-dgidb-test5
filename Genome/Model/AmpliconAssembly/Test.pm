package Genome::Model::AmpliconAssembly::Test;

use strict;
use warnings;

use base 'Test::Class';

use File::Copy 'copy';
use File::Temp 'tempdir';
use Genome::InstrumentData::Sanger::Test;
use Genome::Model::InstrumentDataAssignment;
use Genome::ProcessingProfile::AmpliconAssembly::Test;
use Genome::Utility::FileSystem;
use Test::More;

#< DIR >#
sub test_dir {
    return '/gsc/var/cache/testsuite/data/Genome-Model-AmpliconAssembly';
}

sub _test_dir_for_type {
    die "No type to get test dir\n" if not defined $_[0] or $_[0] eq __PACKAGE__;
    return test_dir().'/'.$_[0];
}

#< MOCK ># 
sub create_mock_model {
    my $self = shift;

    # Processing profile
    my $pp = Genome::ProcessingProfile::AmpliconAssembly::Test->create_mock_processing_profile
        or return;

    #my $data_dir = File::Temp::tempdir(DIR => $self->test_dir, CLEANUP => 0);
    my $data_dir = File::Temp::tempdir(CLEANUP => 1);
    die "Can't make temp data directory for mock model\n" unless -d $data_dir;

    # Model
    my $model = Genome::Model::AmpliconAssembly->create_mock(
        id => -5000,
        genome_model_id => -5000,
        name => 'mr. mock',
        subject_name => 'mock_dna',
        subject_type => 'dna_resource_item_name',
        processing_profile_id => $pp->id,
        processing_profile => $pp,
        data_directory => $data_dir,
    )
        or die "Can't create mock model for amplicon assembly\n";
    for my $pp_param ( Genome::ProcessingProfile::AmpliconAssembly->params_for_class ) {
        $model->set_always($pp_param, $pp->$pp_param);
    }
    for my $dir_type ( Genome::Consed::Directory->directories ) {
        my $dir = $data_dir.'/'.$dir_type; 
        $model->set_always($dir_type, $dir);
        mkdir $dir
            or die "Can't make directory ($dir): $!\n";
    }
    $model->mock(
        'amplicons',
        sub{  
            my $dh = Genome::Utility::FileSystem->open_directory( $model->chromat_dir )
                or return;

            my %amplicons;
            while ( my $scf = $dh->read ) {
                next if $scf =~ m#^\.#;
                $scf =~ s#\.gz##;
                $scf =~ /^(.+)\.[bg]\d+$/
                    or next;
                push @{$amplicons{$1}}, $scf;
            }
            $dh->close;

            return \%amplicons;
        },
    );

    #< FILES >#
    for my $type (qw/ assembly reads processed assembly.confirmed assembly.unconfirmed /) {
        my $method = $type.'_fasta';
        $model->set_always($method, sprintf('%s/%s.%s.fasta', $data_dir, $model->subject_name, $type));
    }
    $model->set_always('metrics_file',  sprintf('%s/%s.metrics.txt', $data_dir, $model->subject_name));

    return $model;
}

sub create_mock_model_with_instrument_data {
    my $self = shift;

    my $model = $self->create_mock_model
        or return;

    my $inst_data = Genome::InstrumentData::Sanger::Test->create_mock_instrument_data; # this dies if no workee
    my $ida = Genome::Model::InstrumentDataAssignment->create_mock(
        id => -5000,
        model_id => $model->id,
        instrument_data_id => $inst_data->id,
        first_build_id => undef,
    )
        or die "Can't create mock instrument data assignment\n";
    $ida->set_always('model_id', $model);
    $ida->set_always('instrument_data', $inst_data);
    $ida->mock(
        'first_build_id', sub { 
            my ($ida, $fbi) = @_;
            $ida->{first_build_id} = $fbi if defined $fbi;
            return $ida->{first_build_id}; 
        },
    );
    $model->set_always('instrument_data', ( $inst_data ));
    $model->set_always('instrument_data_assignments', ( $ida ));

    return $model;
}

#< COPY DATA >#
sub copy_dirs_to_model {
    my ($self, $model, @dirs) = @_;

    die "Need dir types to copy from\n" unless @dirs;

    for my $dir ( @dirs ) {
        my $dest = ( $model->can($dir) )
        ? $model->$dir
        : $model->data_directory;
        $self->_copy_dir(_test_dir_for_type($dir), $dest)
            or return;
    }

    return 1;
}

sub _copy_dir {
    my ($self, $source, $dest) = @_;

    my $dh = Genome::Utility::FileSystem->open_directory($source)
        or return;

    while ( my $file = $dh->read ) {
        next if $file =~ m#^\.#;
        File::Copy::copy("$source/$file", $dest)
            or die "Can't copy ($source/$file) to ($dest): $!\n";
    }

    return 1;

}

#< REAL MODEL FOR TESTING >#

1;

=pod

=head1 Name

ModuleTemplate

=head1 Synopsis

=head1 Usage

=head1 Methods

=head2 

=over

=item I<Synopsis>

=item I<Arguments>

=item I<Returns>

=back

=head1 See Also

=head1 Disclaimer

Copyright (C) 2005 - 2008 Washington University Genome Sequencing Center

This module is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

=head1 Author(s)

B<Eddie Belter> I<ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

