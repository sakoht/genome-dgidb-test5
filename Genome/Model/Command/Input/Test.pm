
###########################################################################

package Genome::Model::Command::Input::Test;

use strict;
use warnings;

use base 'Test::Class';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input';
}



###########################################################################

package Genome::Model::Command::Input::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;
require Genome::Model::Test;
require File::Temp;
require File::Path;

sub valid_param_sets {
    return grep { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub invalid_param_sets {
    return map { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::Test->create_basic_mock_model(type_name => 'tester')
            or confess "Can't create mock tester model.";
    }

    return $self->{_model};
}

sub startup : Tests(startup => 1) {
    my $self = shift;
    
    no warnings 'once';
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    is_deeply(
        [ Genome::Model::Command->get_model_type_names ],
        [qw/ tester /],
        'Set tester type name.',
    );

    $self->_startup or confess "Failed startup.";
    
    return 1;
}

sub _startup {
    return 1;
}

# 
sub __pre_execute {
    my $self = shift;

    print Dumper([$self->_model->inputs]);
    
    return 1;
}

sub __post_execute {
    my $self = shift;

    print Dumper([$self->_model->inputs]);
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Add::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Add';
}

sub _valid_param_sets {
    return (
        { # add multiple 'value_ids' 
            name => 'friends',
            values => 'Watson,Crick',
        },
        { # add single 'value'
            name => 'inst_data',
            values => '2sep09.934pmaa1',
        },
    );
}

sub invalid_param_sets {
    return (
        {# try to add a not 'is_many' input
            name => 'coolness', 
            values => 'none',
        },
        { # try to add Crick again
            name => 'friends',
            values => 'Crick',
        },
        { # try to add id that don't exist
            name => 'inst_data',
            values => '2sep09.934noexist',
        },
    );
}

sub _pre_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @values = $self->_model->$name;
    ok(!@values, "No $name found in model.");
    
    return 1;
}

sub _post_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @values = $self->_model->$name;
    ok(@values, "Added $name in model.");
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Remove::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Remove';
}

sub _valid_param_sets {
    return (
        { # remove multiple 'value_ids' 
            name => 'friends',
            values => 'Watson,Crick',
        },
        { # remove single 'value'
            name => 'inst_data',
            values => '2sep09.934pmaa1',
        },
    );
}

sub invalid_param_sets {
    return (
        {# try to remove a not 'is_many' input
            name => 'coolness', 
            values => 'none',
        },
        { # try to remove input that is not linked to the model anymore
            name => 'inst_data',
            values => '2sep09.934pmaa1',
        },
    );
}

sub friends {
    return 'Crick';
}

sub inst_data {
    return Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
}

sub _startup {
    my $self = shift;

    my $model = $self->_model;

    $model->add_friend('Crick');
    $model->add_friend('Watson');
    my @friends = $model->friends();
    is_deeply(\@friends, [qw/ Crick Watson /], 'Added friends to remove.');
    my $id = Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
        or confess "Can't get sange id 2sep09.934pmaa1";
    $model->add_inst_data($id);
    my @inst_data = $model->inst_data;
    is_deeply(\@inst_data, [ $id ], 'Added instr_data to remove');

    return 1;
}

sub _post_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @values = $self->_model->$name;
    #print Dumper(\@values);
    ok(!@values, "Removed $name from model.");
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::List::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::List';
}

sub valid_param_sets {
    return (
        {
            filter => 'model_id='.$_[0]->_model->id,
        },
    );
}

sub invalid_param_sets {
}

sub _startup {
    my $self = shift;

    $self->_model->coolness('low');
    is($self->_model->coolness, 'low', 'Set coolness (input)');

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Names::Test;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Names';
}

sub valid_param_sets {
    return (
        {
            type_name => 'tester',
        },
    );
}

sub invalid_param_sets {
    return (
        {
            type_name => 'none',
        },
    );
}

sub startup : Tests(startup) {
    my $self = shift;
    
    no warnings 'once';
    no warnings 'redefine';
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Update::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Update';
}

sub _valid_param_sets {
    return (
        { # remove single 'value'
            name => 'coolness',
            value => 'moderate',
        },
        { # remove single 'value'
            name => 'coolness',
            value => 'UNDEF',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to update an 'is_many' input
            name => 'friends',
            value => 'Watson',
        },
    );
}

###########################################################################

1;

=pod

=head1 Tests

=head1 Disclaimer

 Copyright (C) 2009 Washington University Genome Sequencing Center

 This script is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY
 or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
 License for more details.

=head1 Author(s)

 Eddie Belter <ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

