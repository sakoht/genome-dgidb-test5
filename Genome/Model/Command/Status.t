#!/gsc/bin/perl

use strict;
use warnings;

use File::Path;
use Test::More tests => 3;

use above 'Genome';

my $model_id = '2771359026';

my $model_status = Genome::Model::Command::Status->create(genome_model_id=>$model_id, display_output=>0);

ok($model_status);

my $rv = $model_status->execute;

is($rv, 1, 'Testing for successful execution.  Expecting 1.  Got: '.$rv);

my $length_test = 0;

my $xml = $model_status->xml();

if (length($xml) > 3000 ) {
    $length_test = 1 ;
} 

is($length_test,1,'Testing success: Expecting a longish XML string (>3000 chars). Got a string of length: '.length($xml));

