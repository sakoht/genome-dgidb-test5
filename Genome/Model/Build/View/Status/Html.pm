#:boberkfe this looks like a good place to use memcache to cache up some build status.
#:boberkfe when build events update, stuff their status into memcache.  gathering info otherwise
#:boberkfe can get reaaaaal slow.

package Genome::Model::Build::View::Status::Html;

use strict;
use warnings;
use Genome;
use Data::Dumper;
use XML::LibXML;
use XML::LibXSLT;

class Genome::Model::Build::View::Status::Html {
    is => 'UR::Object::View::Default::Html',
    has => [
        _doc    => { 
            is_transient => 1, 
            doc => 'the XML::LibXML document object used to build the content for this view' 
        },
    ],
    has_optional => [
        xsl_file => {
            is => 'Text',
            doc => "Parameter which allows the user to specify the XSL file to transform the output by.",
        },

        # these just pass-through to the underlying XML view
        instance_id => {
            is => 'String',
            doc => 'Optional id of the workflow operation instance to use.'
        },
        use_lsf_file => {
            is => 'Integer',
            default_value => 0,
            doc => "A flag which lets the user retrieve LSF status from a temporary file rather than using a bjobs command to retrieve the values.",
        },
    ],
};

sub _generate_content {
    my $self = shift;

    my $subject = $self->subject;
    return unless $subject;

    # get the xsl
    unless ($self->xsl_file) {
        $self->xsl_file(__FILE__ . '.xsl');
    }
    unless (-e $self->xsl_file) {
        die "Failed to find xsl file: " . $self->xsl_file;
    }
    my $parser = XML::LibXML->new;
    my $xslt = XML::LibXSLT->new;
    my $fh = Genome::Utility::FileSystem->open_file_for_reading($self->xsl_file);
    my $xsl_template = do { local( $/ ) ; <$fh> } ;
    $fh->close();

    # get the xml
    $DB::single = 1;
    my $xml_view = $subject->create_view(
        perspective => $self->perspective,
        toolkit => 'xml',

        # custom for this view
        instance_id => $self->instance_id, 
        use_lsf_file => $self->use_lsf_file, 
    );   
    my $xml_content = $xml_view->_generate_content();

    # convert the xml
    my $source = $parser->parse_string($xml_content);
    my $style_doc = $parser->parse_string($xsl_template);
    my $stylesheet = $xslt->parse_stylesheet($style_doc);
    my $results = $stylesheet->transform($source);
    my $content = $stylesheet->output_string($results);

    return $content;
}

1;

