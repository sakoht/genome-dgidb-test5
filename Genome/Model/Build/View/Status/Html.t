#!/usr/bin/env perl
use strict;
use warnings;

use above "Genome";
use Genome::Model::Build::View::Status::Html; 

use Test::More tests => 4;

# TODO: use one of the test builds
my $subject = Genome::Model::Build->get(101289765);
ok($subject, "found expected build subject") or die "test cannot continue...";

my $view_obj = Genome::Model::Build::View::Status::Html->create(
    subject_id => 101289765,
    #use_lsf_file => 1,
    xsl_root => Genome->base_dir . '/xsl',
    rest_variable => '/cgi-bin/rest.cgi',
    toolkit => 'html',
    perspective => 'status',
); 
ok($view_obj, "created a view") or die "test cannot continue...";

my $html = $view_obj->_generate_content();
ok($html, "view returns HTML") or die "test cannot continue...";

my @diff =
    grep { $_ !~ /generated-at/ }
    grep { /\w/ }
    Genome::Utility::FileSystem->diff_file_vs_text(__FILE__ . '.expected',$html);
    
    SKIP: {
        skip "expected output is broken, will fix by Tuesday 6/1", 1;
        is("@diff","","HTML has no differences from expected value");
    }
