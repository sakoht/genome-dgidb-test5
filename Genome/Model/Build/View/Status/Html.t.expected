<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Status </title>
<link rel="shortcut icon" href="/resources/report_resources/apipe_dashboard/images/gc_favicon.png" type="image/png">
<link rel="stylesheet" href="/resources/report_resources/apipe_dashboard/css/master.css" type="text/css" media="screen">
<link rel="stylesheet" href="/resources/report_resources/apipe_dashboard/css/tablesorter.css" type="text/css" media="screen">
<script type="text/javascript" src="/resources/report_resources/jquery/jquery.js"></script><script type="text/javascript" src="/resources/report_resources/jquery/jquery.tablesorter.min.js"></script>
</head>
<body><div class="container"><div class="background">
<div class="page_header"><table cellpadding="0" cellspacing="0" border="0"><tr>
<td><a href="status.cgi" alt="Go to Search Page" title="Go to Search Page"><img src="/resources/report_resources/apipe_dashboard/images/gc_header_logo2.png" width="44" height="45" align="absmiddle"></a></td>
<td><h1>Analysis Reports v0.2</h1></td>
</tr></table></div>
<div class="page_padding">
<script type="text/javascript">
          
              $(document).ready(function() {
                  $("#alignment").tablesorter({
                      // sort on first column, ascending
                      sortList: [[0,0]]
                  });
              });
          
        </script><script type="text/javascript" src="/resources/report_resources/jquery/boxy/src/javascripts/jquery.boxy.js"></script><link rel="stylesheet" href="/resources/report_resources/jquery/boxy/src/stylesheets/boxy.css" type="text/css">
<script type="text/javascript">
          
                   function event_popup(eventObject) {
                   // assemble event info into a table
                   var popup_content = '<table class="boxy_info" cellpadding="0" cellspacing="0" border="0"><tbody>';
                   for (prop in eventObject) {
                   if (prop != 'popup_title') {
                   popup_content += '<tr><td class="label">' + prop.replace(/_/g," ") + ':</td><td class="value">' + eventObject[prop] + '</td></tr>';
                   }
                   }

popup_content += '</tbody></table>';

// create popup
var popup = new Boxy(popup_content, {title:eventObject.popup_title, fixed:false});
popup.center();
}
          
        </script><h2 class="page_title build_icon">Build 101289765 Status</h2>
<table cellpadding="0" cellspacing="0" border="0" class="info_table_group"><tr>
<td><table border="0" cellpadding="0" cellspacing="0" class="info_table" width="100%">
<tr>
<td class="label">Model ID:</td>
<td class="value"><a xmlns:rest="urn:rest" href="/cgi-bin/rest.cgi/Genome/Model/ReferenceAlignment/status.html?id=2851407278">2851407278</a></td>
</tr>
<tr>
<td class="label">Model Name:</td>
<td class="value">ssmith-test2-realsample-realpp</td>
</tr>
<tr>
<td class="label">Common Name:</td>
<td class="value">AML51</td>
</tr>
<tr>
<td class="label">LSF Job ID:</td>
<td class="value">966891</td>
</tr>
<tr>
<td class="label">Workflow Instance ID:</td>
<td class="value">
<a xmlns:rest="urn:rest" href="/cgi-bin/rest.cgi/Workflow/Store/Db/Model/Instance/status.html?id=1753513">1753513</a> (done)</td>
</tr>
</table></td>
<td><table border="0" cellpadding="0" cellspacing="0" class="info_table" width="100%">
<tr>
<td class="label">Status:</td>
<td class="value">Succeeded <a href="https://gscweb.gsc.wustl.edu//gscmnt/sata427/info/model_data/2851407278/build101289765/reports/Summary/report.html" class="grey">(view build summary report)</a>
</td>
</tr>
<tr>
<td class="label">Processing Profile:</td>
<td class="value">bwa0.5.5 and samtools r453 and picard r107</td>
</tr>
<tr>
<td class="label">Data Directory:</td>
<td class="value"><a href="https://gscweb.gsc.wustl.edu//gscmnt/sata427/info/model_data/2851407278/build101289765">/gscmnt/sata427/info/model_data/2851407278/build101289765</a></td>
</tr>
<tr>
<td class="label">Disk Space (kbytes):</td>
<td class="value">4,984,448</td>
</tr>
<tr>
<td class="label">Error File:</td>
<td class="value"><a href="https://gscweb.gsc.wustl.edu//gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289767.err">101289767.err</a></td>
</tr>
</table></td>
</tr></table>
<table border="0" cellpadding="0" cellspacing="0" class="stages" width="100%"><tr>
<td><table class="stage" border="0" cellpadding="0" cellspacing="0" width="100%">
<tr><th colspan="2">alignment</th></tr>
<tr class="ghost">
<td class="label">
                          Scheduled:
                        </td>
<td class="value">0</td>
</tr>
<tr class="ghost">
<td class="label">
                          Running:
                        </td>
<td class="value">0</td>
</tr>
<tr class="label">
<td class="label">
                          Succeeded:
                        </td>
<td class="value">1</td>
</tr>
<tr class="ghost">
<td class="label">
                          Abandoned:
                        </td>
<td class="value">0</td>
</tr>
<tr class="ghost">
<td class="label">
                          Crashed/Failed:
                        </td>
<td class="value">0</td>
</tr>
<tr class="total">
<td class="label">
                            Total:
                          </td>
<td class="value">1</td>
</tr>
</table></td>
<td><table class="stage" border="0" cellpadding="0" cellspacing="0" width="100%">
<tr><th colspan="2">deduplication</th></tr>
<tr class="ghost">
<td class="label">
                          Scheduled:
                        </td>
<td class="value">0</td>
</tr>
<tr class="ghost">
<td class="label">
                          Running:
                        </td>
<td class="value">0</td>
</tr>
<tr class="label">
<td class="label">
                          Succeeded:
                        </td>
<td class="value">2</td>
</tr>
<tr class="ghost">
<td class="label">
                          Abandoned:
                        </td>
<td class="value">0</td>
</tr>
<tr class="ghost">
<td class="label">
                          Crashed/Failed:
                        </td>
<td class="value">0</td>
</tr>
<tr class="total">
<td class="label">
                            Total:
                          </td>
<td class="value">2</td>
</tr>
</table></td>
<td><table class="stage" border="0" cellpadding="0" cellspacing="0" width="100%">
<tr><th colspan="2">variant detection</th></tr>
<tr class="ghost">
<td class="label">
                          Scheduled:
                        </td>
<td class="value">0</td>
</tr>
<tr class="ghost">
<td class="label">
                          Running:
                        </td>
<td class="value">0</td>
</tr>
<tr class="label">
<td class="label">
                          Succeeded:
                        </td>
<td class="value">2</td>
</tr>
<tr class="ghost">
<td class="label">
                          Abandoned:
                        </td>
<td class="value">0</td>
</tr>
<tr class="ghost">
<td class="label">
                          Crashed/Failed:
                        </td>
<td class="value">0</td>
</tr>
<tr class="total">
<td class="label">
                            Total:
                          </td>
<td class="value">2</td>
</tr>
</table></td>
<td><table class="stage" border="0" cellpadding="0" cellspacing="0" width="100%">
<tr><th colspan="2">generate reports</th></tr>
<tr class="ghost">
<td class="label">
                          Scheduled:
                        </td>
<td class="value">0</td>
</tr>
<tr class="ghost">
<td class="label">
                          Running:
                        </td>
<td class="value">0</td>
</tr>
<tr class="label">
<td class="label">
                          Succeeded:
                        </td>
<td class="value">1</td>
</tr>
<tr class="ghost">
<td class="label">
                          Abandoned:
                        </td>
<td class="value">0</td>
</tr>
<tr class="ghost">
<td class="label">
                          Crashed/Failed:
                        </td>
<td class="value">0</td>
</tr>
<tr class="total">
<td class="label">
                            Total:
                          </td>
<td class="value">1</td>
</tr>
</table></td>
</tr></table>
<hr>
<h2>alignment</h2>
<table class="list tablesorter" width="100%" cellspacing="0" cellpadding="0" border="0" id="alignment">
<colgroup>
<col width="40%">
<col>
<col>
<col>
<col>
</colgroup>
<thead><tr>
<th>flow cell</th>
<th>status</th>
<th>scheduled</th>
<th>completed</th>
<th class="last">elapsed</th>
</tr></thead>
<tbody><tr onmouseover="this.className = 'hover'" onmouseout="this.className=''" onclick='javascript:event_popup({event_id:"101289768",event_status:"Succeeded",instance_id:"1753519",execution_count:"1",instance_status:"done",lsf_job_id:"966892",lsf_job_status:"UNAVAILABLE",date_scheduled:"2010-02-28 22:12:06",date_completed:"2010-02-28 22:36:11",elapsed_time:"24:05",instrument_data_id:"2849448911",output_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/solexa/100109_HWUSI-EAS1605_100737092_61DTD/101289768.out\"&gt;101289768.out&lt;/a&gt;",error_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/solexa/100109_HWUSI-EAS1605_100737092_61DTD/101289768.err\"&gt;101289768.err&lt;/a&gt;",alignment_directory:"&lt;a href=\"http://gscweb/gscmnt/sata912/info/alignment_data/bwa0_5_5/cd6fc20ad94e451c80345e7544d30cc7/NCBI-human-build36/100109_HWUSI-EAS1605_100737092_61DTD/1_2849448911\"&gt;1_2849448911&lt;/a&gt;",project_name:"AML Whole Genome Sequencing",sample_name:"H_KA-943309-S.18828",run_name:"100109_HWUSI-EAS1605_100737092_61DTD",flow_cell_id:"61DTD",read_length:"100",library_name:"H_KA-943309-S.18828-lib1",library_id:"2847322310",subset_name:"1",seq_id:"2849448911",run_type:"Paired",popup_title:"ReferenceAlignment::AlignReads::Bwa #101289768"});'>
<td>61DTD Lane: 1</td>
<td class="status succeeded">succeeded</td>
<td>2010-02-28 22:12:06</td>
<td>2010-02-28 22:36:11</td>
<td class="last">24:05</td>
</tr></tbody>
</table>
<h2>deduplication</h2>
<table class="list tablesorter" width="100%" cellspacing="0" cellpadding="0" border="0" id="deduplication">
<colgroup>
<col width="40%">
<col>
<col>
<col>
<col>
</colgroup>
<thead><tr>
<th>event</th>
<th>status</th>
<th>scheduled</th>
<th>completed</th>
<th class="last">elapsed</th>
</tr></thead>
<tbody>
<tr onmouseover="this.className = 'hover'" onmouseout="this.className=''" onclick='javascript:event_popup({event_id:"101289769",event_status:"Succeeded",instance_id:"1753524",execution_count:"1",instance_status:"done",lsf_job_id:"966931",lsf_job_status:"UNAVAILABLE",date_scheduled:"2010-02-28 22:36:33",date_completed:"2010-02-28 23:14:48",elapsed_time:"38:15",instrument_data_id:"",output_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289769.out\"&gt;101289769.out&lt;/a&gt;",error_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289769.err\"&gt;101289769.err&lt;/a&gt;",alignment_directory:"",popup_title:"ReferenceAlignment::DeduplicateLibraries::Picard #101289769"});'>
<td>ReferenceAlignment::DeduplicateLibraries::Picard</td>
<td class="status succeeded">succeeded</td>
<td>2010-02-28 22:36:33</td>
<td>2010-02-28 23:14:48</td>
<td class="last">38:15</td>
</tr>
<tr onmouseover="this.className = 'hover'" onmouseout="this.className=''" onclick='javascript:event_popup({event_id:"101289770",event_status:"Succeeded",instance_id:"1753525",execution_count:"1",instance_status:"done",lsf_job_id:"966951",lsf_job_status:"UNAVAILABLE",date_scheduled:"2010-02-28 23:15:02",date_completed:"2010-02-28 23:15:10",elapsed_time:"00:08",instrument_data_id:"",output_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289770.out\"&gt;101289770.out&lt;/a&gt;",error_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289770.err\"&gt;101289770.err&lt;/a&gt;",alignment_directory:"",popup_title:"ReferenceAlignment::PostDedupReallocate #101289770"});'>
<td>ReferenceAlignment::PostDedupReallocate</td>
<td class="status succeeded">succeeded</td>
<td>2010-02-28 23:15:02</td>
<td>2010-02-28 23:15:10</td>
<td class="last">00:08</td>
</tr>
</tbody>
</table>
<h2>variant detection</h2>
<table class="list tablesorter" width="100%" cellspacing="0" cellpadding="0" border="0" id="variant_detection">
<colgroup>
<col width="40%">
<col>
<col>
<col>
<col>
</colgroup>
<thead><tr>
<th>event</th>
<th>status</th>
<th>scheduled</th>
<th>completed</th>
<th class="last">elapsed</th>
</tr></thead>
<tbody>
<tr onmouseover="this.className = 'hover'" onmouseout="this.className=''" onclick='javascript:event_popup({event_id:"101289771",event_status:"Succeeded",instance_id:"1753530",execution_count:"1",instance_status:"done",lsf_job_id:"966953",lsf_job_status:"UNAVAILABLE",date_scheduled:"2010-02-28 23:15:26",date_completed:"2010-02-28 23:15:27",elapsed_time:"00:01",instrument_data_id:"",output_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/all_sequences/101289771.out\"&gt;101289771.out&lt;/a&gt;",error_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/all_sequences/101289771.err\"&gt;101289771.err&lt;/a&gt;",alignment_directory:"",popup_title:"ReferenceAlignment::UpdateGenotype::Samtools #101289771"});'>
<td>ReferenceAlignment::UpdateGenotype::Samtools</td>
<td class="status succeeded">succeeded</td>
<td>2010-02-28 23:15:26</td>
<td>2010-02-28 23:15:27</td>
<td class="last">00:01</td>
</tr>
<tr onmouseover="this.className = 'hover'" onmouseout="this.className=''" onclick='javascript:event_popup({event_id:"101289772",event_status:"Succeeded",instance_id:"1753531",execution_count:"1",instance_status:"done",lsf_job_id:"966970",lsf_job_status:"UNAVAILABLE",date_scheduled:"2010-02-28 23:15:43",date_completed:"2010-03-01 00:01:48",elapsed_time:"46:05",instrument_data_id:"",output_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/all_sequences/101289772.out\"&gt;101289772.out&lt;/a&gt;",error_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/all_sequences/101289772.err\"&gt;101289772.err&lt;/a&gt;",alignment_directory:"",popup_title:"ReferenceAlignment::FindVariations::Samtools #101289772"});'>
<td>ReferenceAlignment::FindVariations::Samtools</td>
<td class="status succeeded">succeeded</td>
<td>2010-02-28 23:15:43</td>
<td>2010-03-01 00:01:48</td>
<td class="last">46:05</td>
</tr>
</tbody>
</table>
<h2>generate reports</h2>
<table class="list tablesorter" width="100%" cellspacing="0" cellpadding="0" border="0" id="generate_reports">
<colgroup>
<col width="40%">
<col>
<col>
<col>
<col>
</colgroup>
<thead><tr>
<th>event</th>
<th>status</th>
<th>scheduled</th>
<th>completed</th>
<th class="last">elapsed</th>
</tr></thead>
<tbody><tr onmouseover="this.className = 'hover'" onmouseout="this.className=''" onclick='javascript:event_popup({event_id:"101289773",event_status:"Succeeded",instance_id:"1753536",execution_count:"1",instance_status:"done",lsf_job_id:"966985",lsf_job_status:"UNAVAILABLE",date_scheduled:"2010-03-01 00:02:00",date_completed:"2010-03-01 07:14:30",elapsed_time:"07:12:30",instrument_data_id:"",output_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289773.out\"&gt;101289773.out&lt;/a&gt;",error_log_file:"&lt;a href=\"http://gscweb/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289773.err\"&gt;101289773.err&lt;/a&gt;",alignment_directory:"",popup_title:"ReferenceAlignment::RunReports #101289773"});'>
<td>ReferenceAlignment::RunReports</td>
<td class="status succeeded">succeeded</td>
<td>2010-03-01 00:02:00</td>
<td>2010-03-01 07:14:30</td>
<td class="last">07:12:30</td>
</tr></tbody>
</table>
</div>
</div></div></body>
</html>
